ksqldb-tools/src/main/java/io/confluent/ksql/tools/migrations/commands/ApplyMigrationCommand.java

```diff
@@ -240,6 +240,11 @@ private boolean apply(
       if (!migration.isPresent()) {
         throw new MigrationException("No migration file with version " + version + " exists.");
       }
+      if (version < minimumVersion) {
+        throw new MigrationException(
+            "Version must be newer than the last version migrated. Last version migrated was "
+                + (minimumVersion - 1));
```

Comment:
nit: it's not great that `apply()` gets the latest migrated version, transforms that into `minimumVersion` to pass to `loadMigrationsToApply()`, and then `loadMigrationsToApply()` undoes the transformation again in this error message. Maybe we can have `apply()` directly pass the last migrated version and move the logic for computing `minimumVersion` into `loadMigrationsToApply()` instead? Less coupling between the two methods that way.

Reply:
updated